### PR TITLE
#trivial MOBILE-1039 Fix invalid state when next is pressed before the evaluate-context run

### PR DESCRIPTION
**Pull request by phcp**

### CODE REVIEW

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/DDMFormPresenter.kt [hunk 1]
```diff
@@ -44,6 +44,13 @@ class DDMFormPresenter(val view: DDMFormViewContract.DDMFormView) : DDMFormViewC
 			val isValid = fieldContext.isValid ?: true
 
 			field.lastValidationResult = isValid
+
+			if (!isValid) {
```
Comment (victorlaerte) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/DDMFormPresenter.kt]: Maybe invert the order is better. `if (isValid) else`

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/DDMFormPresenter.kt [hunk 2]
```diff
@@ -85,26 +92,30 @@ class DDMFormPresenter(val view: DDMFormViewContract.DDMFormView) : DDMFormViewC
...
-		if (!field.isTransient) {
+	override fun syncForm(formInstance: FormInstance, field: Field<*>?, onComplete: (() -> Unit)?) {
+		val fieldIsTransient = field?.isTransient ?: false
+		val invalidStateForSync = formInstanceState != FormInstanceState.IDLE
 
-			if (!view.hasConnectivity()) {
-				view.showOfflineWarningMessage()
-				return
-			}
+		if (fieldIsTransient || invalidStateForSync) return
```
Comment (victorlaerte) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/DDMFormPresenter.kt]: a typical swift guard let 😛

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/DDMFormPresenter.kt [hunk 3]
```diff
@@ -114,7 +125,10 @@ class DDMFormPresenter(val view: DDMFormViewContract.DDMFormView) : DDMFormViewC
 	}
 
 	override fun submit(formInstance: FormInstance, isDraft: Boolean) {
-		if (isDraft && !view.config.autosaveDraftEnabled) return
+		val autosaveDisabled = isDraft && !view.config.autosaveDraftEnabled
+		val invalidStateForSubmission = formInstanceState != FormInstanceState.IDLE
+
+		if (autosaveDisabled || invalidStateForSubmission) return
```
Comment (victorlaerte) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/DDMFormPresenter.kt]: 👍

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/DDMFormPresenter.kt (hunk 1 above)
Comment (phcp) [reply android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/DDMFormPresenter.kt]: Yeah, that's true.

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/DDMFormPresenter.kt (hunk 2 above)
Comment (phcp) [reply android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/DDMFormPresenter.kt]: kkkkkk swift feelings 😆

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/DDMFormPresenter.kt (hunk 1 above)
Comment (phcp) [reply android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/DDMFormPresenter.kt]: Fixed

**Pull request merged**